src/interfaces/simulation.ts

```diff
@@ -136,57 +171,68 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     toToken: Address,
     options: SimulationOptions = {}
   ): Promise<TransactionOutcome> {
+    const withdrawArgs = { from, fromVault, amount, toToken, options };
+
     const signer = this.ctx.provider.write.getSigner(from);
     const vaultContract = new YearnVaultContract(fromVault, signer);
-    const underlyingToken = await vaultContract.token();
-    const isZapping = underlyingToken !== getAddress(toToken);
-    let forkId: string | undefined;
-    let simulateWithdrawal: (save: boolean) => Promise<TransactionOutcome>;
-
-    if (isZapping) {
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
-      let needsApproving: boolean;
 
-      if (fromVault === EthAddress) {
-        needsApproving = false;
-      } else {
-        needsApproving = await this.yearn.services.zapper
-          .zapOutApprovalState(from, fromVault)
-          .then((state) => !state.isApproved)
-          .catch(() => {
-            throw new ZapperError("zap out approval state", ZapperError.ZAP_OUT_APPROVAL_STATE);
-          });
-      }
+    const [vault] = await this.yearn.vaults.get([fromVault]);
```

Comment:
not for this PR specifically but we should have at some point a specific endpoint to fetch just one vault instead of having to fetch them all and the filter through the entire array

Reply:
Agreed, I started doing that for example in the `src/services/meta.ts`
```typescript
async vault(address: Address) {}
```